### PR TITLE
Standarize Settings, Params and Parameters in Receiver

### LINKED ISSUE #2650: Standardize on params vs settings vs options
In few places for constructors we use "Params" (in the components), some "Parameters" (in the Service) or "Settings" in the components helpers like exporterhelper.

We need to decide on a name a pattern: expose a struct with these properties or expose "Options" and use variadic args.

Consolidate all interfaces to follow the same pattern.

### PR DESCRIPTION
**Pull request by pmatyjasek-sumo**
**Description:** 
Continuation work from #2991
Fixes #2650
Replace ReceiverCreateParams with ReceiverCreateSettings.
Replace all dependencies in Receivers.

**Link to tracking Issue:** 
#2650 

**Testing:**
Unit tests

**Documentation:**

### CODE REVIEW

File: component/componenttest/nop_receiver_test.go [hunk 1]
```diff
@@ -33,17 +33,17 @@ func TestNewNopReceiverFactory(t *testing.T) {
 	cfg := factory.CreateDefaultConfig()
 	assert.Equal(t, &nopReceiverConfig{ReceiverSettings: config.NewReceiverSettings(config.NewID("nop"))}, cfg)
 
-	traces, err := factory.CreateTracesReceiver(context.Background(), component.ReceiverCreateParams{}, cfg, consumertest.NewNop())
+	traces, err := factory.CreateTracesReceiver(context.Background(), component.ReceiverCreateSettings{}, cfg, consumertest.NewNop())
```
Comment (carlosalberto) [inline component/componenttest/nop_receiver_test.go:36]: Just as the other test, it may be useful to have a nop instance. Not a biggie though.

**Review by carlosalberto — APPROVED** [review]

File: component/componenttest/nop_receiver_test.go (hunk 1 above)
Comment (pmatyjasek-sumo) [reply component/componenttest/nop_receiver_test.go:36]: Sure, but I didn't want to change any logic there. Only refactor the code

**Review by bogdandrutu — APPROVED** [review]
Please rebase

**Comment by bogdandrutu** [comment]
Lint errors

**Comment by bogdandrutu** [comment]
Failing tests not fixed.

**Comment by pmatyjasek-sumo** [comment]
Rebased and tests fixed

**Comment by bogdandrutu** [comment]
Sorry, merged another one of our PRs and a rebase is needed again :)

**Comment by pmatyjasek-sumo** [comment]
ok :) I'll do it again

**Pull request merged**